File: ws/topon/U2/Ws19_Menu/ContactsBookMenu/src/main/java/utils/MongoDbManager.java
package utils;

/**
 *
 * @author Mateo Topon
 */
public class MongoDbManager {
    
    public static void connect(String connectionString){
        System.out.println("connecting");
        
    }

    public static boolean add(Object object) {
        String collection=null;
        boolean saved = false;
        try {
            collection = object.getClass().toString();
            System.out.println("saving -->" + collection + "<--" + object);
            saved = true;


        } catch (Exception ex) {
            System.out.println("error on saving" + collection +"");
            saved = false;
        }
        return saved;
    }

}

File: EXAM/jami/u3/StrategyPattern/utils/MongoDBManager.py
import pymongo
from pymongo import MongoClient

class MongoDBManager:
    def __init__(self):
        self.connectionString = "mongodb+srv://kleverdavid165:POO2025@cluster0.0hjyg.mongodb.net/"
        self.databaseName = "JamiStrategy"
        self.client = None
        self.db = None
        self.collection = None
        self.connect()
        
    def connect(self):
        try:
            self.client = MongoClient(self.connectionString)
            self.db = self.client[self.databaseName]
            self.collection = self.db['JamiSortedArray']
            print("Connection to MongoDB established successfully")
        except Exception as e:
            print(f"Error connecting to MongoDB: {e}")
            raise RuntimeError("Could not connect to the database")
    
    def saveSortResult(self, unsorted, size, algorithm, sortedList):
        try:
            data = {
                "unsorted": unsorted,
                "size": size, 
                "sort algorithm": algorithm,
                "InsertionSort": sortedList,
                "sorted": sortedList
            }
            self.collection.insert_one(data)
            print("Data stored in MongoDB")
        except Exception as e:
            print(f"Error saving to MongoDB: {e}")
    
    def closeConnection(self):
        if self.client:
            self.client.close()
